### PR TITLE
Add analyzer rules to track allowed category and rule IDs for analyze…

### LINKED ISSUE #870: Rationalize the diagnostic IDs for rules in refactored packages
For example, we have moved out the rules from Roslyn.Diagnostics.Analyzers package to multiple other packages. For some of them, we have chosen an arbitrary CAXXXX ID and for others we have retained the original RSXXXX IDs.

We need to take a pass over all packages, and assign specific CA ID ranges to packages so we avoid duplicates, have a proper range ID for future rules - I am assuming that in future we will add lot of new rules which didn't exist in legacy FXCop.

### PR DESCRIPTION
**Pull request by mavasani**
…rs via an additional file.

Also executed this analyzer on our repo and fixed the rules which were using incorrect range/IDs. We now have a separate file that enforces the allowed categories and corresponding rule ID range for all analyzers in the repo: https://github.com/mavasani/roslyn-analyzers/blob/3dd98c2cd293b2b4e2b250844c1f8c6045c91c5a/src/Utilities/DiagnosticCategoryAndIdRanges.txt

Fixes #870

### CODE REVIEW

File: src/Microsoft.CodeAnalysis.Analyzers/Core/AnalyzerDiagnosticCategory.cs [hunk 1]
```diff
@@ -1,12 +0,0 @@
-﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
```
Comment (mavasani) [inline src/Microsoft.CodeAnalysis.Analyzers/Core/AnalyzerDiagnosticCategory.cs]: FYI: Large number of file changes here are due to the fact that our category strings and resources were duplicated and spread across projects. Now all of them have been unified into the AnalyzerUtilities project.

File: src/Utilities/DiagnosticCategory.cs [hunk 2]
```diff
@@ -13,7 +13,21 @@ internal static class DiagnosticCategory
         public static readonly string Security = AnalyzerUtilitiesResources.CategorySecurity;
         public static readonly string Usage = AnalyzerUtilitiesResources.CategoryUsage;
         public static readonly string Naming = AnalyzerUtilitiesResources.CategoryNaming;
-        public static readonly string ApiDesign = AnalyzerUtilitiesResources.CategoryApiDesign;
         public static readonly string Library = AnalyzerUtilitiesResources.CategoryLibrary;
+        public static readonly string Documentation = AnalyzerUtilitiesResources.CategoryDocumentation;
```
Comment (mavasani) [inline src/Utilities/DiagnosticCategory.cs]: Also filed https://github.com/dotnet/roslyn-analyzers/issues/1468 to track why we use resource strings for Category in all our FxCop analyzers, but use string constants in Roslyn and MS.CA analyzers.

**Comment by mavasani** [comment]
Ping, need a review for this to ensure #870 is fixed for the VSIX release

File: src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticDescriptorCreationAnalyzer.cs [hunk 3]
```diff
@@ -105,5 +255,376 @@ private static void AnalyzeHelpLinkUri(OperationAnalysisContext operationAnalysi
...
+                            {
+                                Debug.Assert(allowedIds.prefix != null);
+
+                                if (ruleId.StartsWith(allowedIds.prefix, StringComparison.Ordinal))
+                                {
+                                    if (ShouldValidateRange(allowedIds))
+                                    {
+                                        var suffix = ruleId.Substring(allowedIds.prefix.Length);
+                                        int ruleIdInt;
+                                        if (int.TryParse(suffix, out ruleIdInt) &&
```
Comment (333fred) [inline src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticDescriptorCreationAnalyzer.cs]: our var :)

**Comment by jinujoseph** [comment]
sorry for the delay in review and thank you for taking care of it

**Pull request merged**